samplePDF/samplePDFFDBase.cpp

```diff
@@ -1497,27 +1532,96 @@ double samplePDFFDBase::GetLikelihood()
   }
   return negLogL;
 }
- 
-#ifndef USE_PROB3 
-// ************************************************
-// Switch from MaCh3 CUDAProb flavour to CUDAProb Probtype
-inline cudaprob3::ProbType samplePDFFDBase::SwitchToCUDAProbType(CUDAProb_nu CUDAProb_nu) {
-//*************************************************  
-  switch(CUDAProb_nu)
-  {
-    case CUDAProb_nu::e_e : return cudaprob3::ProbType::e_e;
-    case CUDAProb_nu::e_m : return cudaprob3::ProbType::e_m;
-    case CUDAProb_nu::e_t : return cudaprob3::ProbType::e_m;
-    case CUDAProb_nu::m_e : return cudaprob3::ProbType::m_e;
-    case CUDAProb_nu::m_m : return cudaprob3::ProbType::m_m;
-    case CUDAProb_nu::m_t : return cudaprob3::ProbType::m_t;
-    case CUDAProb_nu::t_e : return cudaprob3::ProbType::t_e;
-    case CUDAProb_nu::t_m : return cudaprob3::ProbType::t_m;
-    case CUDAProb_nu::t_t : return cudaprob3::ProbType::t_t;
-    default:
-      MACH3LOG_ERROR("Unknown CUDAProbType!");
-      throw;
+
+void samplePDFFDBase::InitialiseSingleFDMCObject(int iSample, int nEvents_) {
+  if (iSample < 0 || iSample >= nSamples) {
+    MACH3LOG_ERROR("Invalid iSample index in InitialiseSingleFDMCObject");
+    MACH3LOG_ERROR("Index given is {} and only {} samples found", iSample, nSamples);
```

Comment:
can we also throw this away:

https://github.com/mach3-software/MaCh3/blob/feature_OscClassMagic/samplePDF/Structs.h#L513-L581